src/lib/merkle_address/merkle_address.ml

```diff
@@ -307,11 +307,11 @@ struct
             ()
         | Some addr' ->
             [%test_result: t option] ~expect:(Some address) (prev addr') )
-end
+end*)
 
 let%test_module "Address" =
   ( module struct
-    module Test4 = Make_test (struct
+    (* module Test4 = Make_test (struct
```

Comment:
I don't think these modules should be removed: while they are not referenced anywhere else, the tests are run as side effects of the functor application (see documentation for `ppx_inline_test`).

Instead, either the warning should be silenced locally for such test modules, or they should be anonymous.